workspace/pkgs-graph/src/index.ts

```diff
@@ -120,3 +126,21 @@ function createPkgMap (pkgs: Package[]): Record<string, Package> {
   }
   return pkgMap
 }
+
+function getPkgMapByManifestName (pkgMapValues: Package[]) {
+  const pkgMapByManifestName: Record<string, Package[] | undefined> = {}
+  for (const pkg of pkgMapValues) {
+    if (pkg.manifest.name) {
+      (pkgMapByManifestName[pkg.manifest.name] ??= []).push(pkg)
+    }
+  }
+  return pkgMapByManifestName
+}
+
+function getPkgMapByDir (pkgMapValues: Package[]) {
+  const pkgMapByDir: Record<string, Package | undefined> = {}
+  for (const pkg of pkgMapValues) {
+    pkgMapByDir[path.resolve(pkg.dir)] = pkg
```

Comment:
This resolve is defensive; since we're comparing resolved paths above, I wanted to ensure everything went through this algorithm before comparison. It's possible that this is not needed, however, there doesn't seem to be any negative performance hit.